### PR TITLE
Examples: Add demo for indexed BufferGeometry

### PR DESCRIPTION
**Pull request by Mugen87**
As requested in #13259

Also correct the docs.

### CODE REVIEW

File: examples/webgl_buffergeometry_indexed.html [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+				camera = new THREE.PerspectiveCamera( 27, window.innerWidth / window.innerHeight, 1, 3500 );
+				camera.position.z = 64;
+
+				scene = new THREE.Scene();
+				scene.background = new THREE.Color( 0x050505 );
+
+				//
+
+				var ambientLight = new THREE.AmbientLight( 0xcccccc );
```
Comment (WestLangley) [inline examples/webgl_buffergeometry_indexed.html]: Ambient light should never be that high. Try 0x222222.

File: examples/webgl_buffergeometry_indexed.html [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+				scene.add( ambientLight );
+
+				var light1 = new THREE.DirectionalLight( 0xffffff, 0.5 );
+				light1.position.set( 1, 1, 1 );
+				scene.add( light1 );
+
+				var light2 = new THREE.DirectionalLight( 0xffffff, 1.5 );
```
Comment (WestLangley) [inline examples/webgl_buffergeometry_indexed.html]: Try not not over-brighten you scene.

File: examples/webgl_buffergeometry_indexed.html [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+				//
+
+				geometry.setIndex( indices );
+				geometry.addAttribute( 'position', new THREE.Float32BufferAttribute( vertices, 3 ) );
+				geometry.addAttribute( 'normal', new THREE.Float32BufferAttribute( normals, 3 ) );
+				geometry.addAttribute( 'color', new THREE.Float32BufferAttribute( colors, 3 ) );
+
+				var material = new THREE.MeshPhongMaterial( {
+					specular: 0xffffff, shininess: 250,
```
Comment (WestLangley) [inline examples/webgl_buffergeometry_indexed.html]: If the specular reflectance of the material is `0xffffff`, and the diffuse reflectance is greater than zero, then more than 100% of the light is being reflected.  Set `specular` to `0x111111`. Study the hot-spot before and after.

Comment (Mugen87) [reply examples/webgl_buffergeometry_indexed.html]: Okay done 👍

Comment (Mugen87) [reply examples/webgl_buffergeometry_indexed.html]: BTW: I've just copied the material from an other example. We might want to review the lightings in the examples at some point...

Comment (WestLangley) [reply examples/webgl_buffergeometry_indexed.html]: Right. Check the outline pass demo. :/

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**